src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala

```diff
@@ -0,0 +1,196 @@
+package org.broadinstitute.hail.utils
+
+import java.util.Locale
+
+import org.broadinstitute.hail.Utils._
+
+object StringEscapeUtils {
+
+  def hex(ch: Char): String = Integer.toHexString(ch).toUpperCase(Locale.ENGLISH)
+
+  def escapeUnprintable(str: String): String = escapeUnprintable(str, new StringBuilder(capacity = str.length * 2))
+
+  def escapeUnprintable(str: String, sb: StringBuilder): String = {
+    sb.clear()
+
+    var sz: Int = 0
+    sz = str.length
+    var i: Int = 0
+    while (i < sz) {
+      {
+        val ch: Char = str.charAt(i)
+        if (ch > 0xfff) {
+          sb.append("\\u" + hex(ch))
+        } else if (ch > 0xff) {
+          sb.append("\\u0" + hex(ch))
+        } else if (ch > 0x7f) {
+          sb.append("\\u00" + hex(ch))
+        } else if (ch < 32) {
+          ch match {
+            case '\b' =>
+              sb += '\\'
+              sb += 'b'
+            case '\n' =>
+              sb += '\\'
+              sb += 'n'
+            case '\t' =>
+              sb += '\\'
+              sb += 't'
+            case '\f' =>
+              sb += '\\'
+              sb += 'f'
+            case '\r' =>
+              sb += '\\'
+              sb += 'r'
+            case _ =>
+              if (ch > 0xf) {
+                sb.append("\\u00" + hex(ch))
```

Comment:
On reflection, I think non-printable characters < 0x7f should be printed in octal: `\xxx`, where xxx is the 3-digit octal representation.